### PR TITLE
Fix and refactor prometheus metrics setup

### PR DESCRIPTION
**Pull request by oliver-nyt**
As discussed with @jprobinson - this fixes Prometheus metrics in gizmo by
- not using invalid metric names but using the standard prometheus.InstrumentHandler() that will generate http_* metrics and use the full endpoint pathname as a label.
- expose prometheus metrics at `/metrics`

🍺

### CODE REVIEW

**Comment by jprobinson** [comment]
It looks like `go get` may have bombed in travis. re-running the job now to verify.

**Comment by jprobinson** [comment]
Ah! This is referencing some GCP dependencies that were removed yesterday, which is why the `go get` is failing...they aren't there :)

File: server/simple_server.go [hunk 1]
```diff
@@ -306,16 +304,15 @@ func (s *SimpleServer) Register(svcI Service) error {
 		// register all context endpoints with our wrapper
 		for path, epMethods := range mcs.JSONEndpoints() {
 			for method, ep := range epMethods {
-				endpointName := metricName(prefix, path, method)
+				fullPath := strings.TrimPrefix(prefix+path, "/")
```
Comment (jprobinson) [inline server/simple_server.go]: We do this `TrimPrefix` a lot. Any chance we can encapsulate it somewhere to avoid the repetition?

Comment (oliver-nyt) [inline server/simple_server.go]: Dunno, there's already a lot of repetition in the `Register()` functions. (something I'd love to get rid of / improve if you have suggestions).

I don't really ned the prefix in `TimedAndCounted()` but I could pass it anyway and do the concatenate and trim in there. Thoughts?

Comment (jprobinson) [inline server/simple_server.go]: Maybe we still do the `prefix+path` when we call `TimedAndCounted`, but then do the `TrimPrefix` inside the `TimedAndCounted`?

(and yes, I agree the register repetition needs to go, but haven't thought of a good strategy yet. Ideas are welcome!)

Comment (oliver-nyt) [inline server/simple_server.go]: Yeah, you're right, the TrimPrefix is only used for the metrics and not the handler so I'll move that into metrics.go.

**Comment by jprobinson** [comment]
🍕 🎉

**Comment by oliver-nyt** [comment]
🌴

**Pull request merged**